.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,226 @@
+name: Setup build
+description: Sets up the build environment for the current job
+
+inputs:
+  windows-sdk-version:
+    description: The Windows SDK version to use, e.g. "10.0.22621.0"
+    required: false
+    type: string
+  windows-msvc-version:
+    description: The Windows MSVC version to use, e.g. "14.42"
+    required: false
+    type: string
+  setup-vs-dev-env:
+    description: Whether to set up a Visual Studio Dev Environment
+    default: false
+    required: false
+    type: boolean
+  target-arch:
+    description: The target architecture, "x86", "amd64" or "arm64". Defaults to the host architecture.
+    required: false
+    type: string
+
+runs:
+  using: composite
+  steps:
+    - name: Verify input
+      id: verify-input
+      shell: pwsh
+      run: |
+        if ($IsWindows) {
+          $HostOs = "windows"
+        } elseif ($IsMacOS) {
+          $HostOs = "mac"
+        } else {
+          Write-Output "::error::Unsupported host OS."
+          exit 1
+        }
+
+        $Arch = ([System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture).ToString()
+        switch ($Arch) {
+          "X64" { $HostArch = "amd64" }
+          "Arm64" { $HostArch = "arm64" }
+          default {
+            Write-Output "::error::Unsupported host architecture: `"$HostArch`""
+            exit 1
+          }
+        }
+
+        $WinMsvcVersion = "${{ inputs.windows-msvc-version }}"
+        if ($WinMsvcVersion -ne "") {
+          $ParsedWinMsvcVersion = [System.Version]::Parse($WinMsvcVersion)
+          if ($ParsedWinMsvcVersion -eq $null) {
+            Write-Output "::error::Invalid Windows MSVC version: `"${WinMsvcVersion}`"."
+            exit 1
+          }
+          if ($ParsedWinMsvcVersion.Major -ne 14) {
+            Write-Output "::error::Unsupported Windows MSVC version (major version not supported): `"${WinMsvcVersion}`"."
+            exit 1
+          }
+          if ($ParsedWinMsvcVersion.Build -ne -1) {
```

Comment:
I'm not sure I follow this. This is the third component, why does it need to be `1`?

Reply:
Can we add a comment explaining that?

Reply:
Done!